### PR TITLE
Add support for building with a wheel and cleanup setup.py

### PR DESCRIPTION
**Pull request by mattrobenolt**
```
$ make publish
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
 
-.PHONY: test
 test:
 	tox
+
+publish:
+	pip install wheel twine
+	rm -rf dist/ build/
+	python setup.py sdist bdist_wheel
+	twine upload dist/*
```
Comment (mattrobenolt) [inline Makefile]: `twine` uploads to pypi over https. `setuptools` uses http.

File: setup.py [hunk 2]
```diff
@@ -5,22 +6,30 @@
...
+        'Development Status :: 5 - Production/Stable',
+        'Intended Audience :: Developers',
+        'Intended Audience :: System Administrators',
+        'License :: OSI Approved :: MIT License',
+        'Operating System :: OS Independent',
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.4',
+        'Programming Language :: Python',
```
Comment (mattrobenolt) [inline setup.py]: Wasn't entirely sure what all versions of python are covered, but 2.7 and 3.4 are tested against at least.

**Comment by solarkennedy** [comment]
Looks fine.
Longer term, what do you think about using tox, like `tox -e package` so that the needed requirements are handled in a virtualenv? (so that you don't have to install anything with pip globally)

**Comment by mattrobenolt** [comment]
I normally do everything inside a virtualenv and don't install things globally anyway, but I don't have much experience with tox personally, so I can't comment. I just use virtualenvwrapper for all my needs.

**Comment by mattrobenolt** [comment]
Also, to be clear, feel free to remove the `pip install` lines. :) Those were added just to assert that those dependencies are needed to publish this way. They could conversely be moved to a requirements-dev.txt or something of the like.

**Pull request merged**